Tests/SwiftLintFrameworkTests/TypeNameRuleTests.swift

```diff
@@ -48,4 +48,18 @@ class TypeNameRuleTests: XCTestCase {
 
         verifyRule(description, ruleConfiguration: ["validates_start_with_lowercase": false])
     }
+
+    func testTypeNameWithLeadingUnderscoreWithPrivate() {
+        let baseDescription = TypeNameRule.description
+
+        let description = baseDescription
+            .with(nonTriggeringExamples: [
+                Example("private class _MyPrivateClass {}", excludeFromDocumentation: true)
+            ])
+            .with(triggeringExamples: [
+                Example("private class ↓`_` {}", excludeFromDocumentation: true)
+            ])
+
+        verifyRule(description)
+    }
```

Comment:
The test cases in this file are actually only there to test special cases involving non-default configuration parameters. Since your tests do not require configuration changes I suggest to put them into the lists of `nonTriggeringExamples` and `triggeringExamples` located in the implementation class itself. It's preferred to have the test cases as close to the rule logic as possible.

Please add a test for `GenericTypeNameRule` too. For `TypeNameRule` the test cases are extracted into `TypeNameRuleExamples`.

Reply:
> The test cases in this file are actually only there to test special cases involving non-default configuration parameters. Since your tests do not require configuration changes I suggest to put them into the lists of nonTriggeringExamples and triggeringExamples located in the implementation class itself. It's preferred to have the test cases as close to the rule logic as possible.

@SimplyDanny Oh... Okay. I'll move it.

> Please add a test for GenericTypeNameRule too. For TypeNameRule the test cases are extracted into TypeNameRuleExamples.

@SimplyDanny I thought about that, GenericTypeNameRule doesn't have NameStripping or anything else, so will be fine. And I think I should revert GenericTypeNameRule's changes.

Reply:
That's right. An anonymous generic type would not make any sense and is therefore not even allowed by the compiler. Good catch!